### PR TITLE
Plink 1.9

### PR DESCRIPTION
**Pull request by astrovsky01**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)
Adding plink 1.9, which is currently more content-complete than Plink2

### CODE REVIEW

File: tools/plink/plink.xml [hunk 1]
```diff
@@ -0,0 +1,1241 @@
+<tool id='plink2' name='plink' version='@TOOL_VERSION@+galaxy@VERSION_SUFFIX@'>
```
Comment (bgruening) [inline tools/plink/plink.xml:1]: let's use the ID `plink` here and later in the pllink2 tool

File: tools/plink/plink.xml [hunk 2]
```diff
@@ -0,0 +1,1241 @@
...
+            $functions.extra_chromosomes
+
+            #if $functions.autosome != 'none'
+                $functions.autosome
+            #end if
+
+        ##SNP specificity
+            #if $functions.snps_exclusives != 'No':
+                --snps-only
+                #if $functions.snps_exclusives == 'acgt':
```
Comment (bgruening) [inline tools/plink/plink.xml:102]: indentation ...

File: tools/plink/plink.xml [hunk 3]
```diff
@@ -0,0 +1,1241 @@
...
+                    </when>
+                    <when value='extract-col-cond-max'>
+                    </when>
+                </conditional> -->
+                <param name='var_min_qual' type='integer' label='Minimum quality score' help='Causes all variants with QUAL value smaller than the given number, or with no QUAL value at all, to be skipped.' optional='true'/>
+                <param name='var_filter' type='text' label='Multiple filters' optional='true' help='Space separated list of filters. Skips varians which fail these filters.'/>
+                <!-- <section name='info'>
+
+                </section> -->
+                <param name='chromosome' type='text' label='Chromosome(s)' help='Excludes all variants not on the listed chromosome(s). Can be listed as single chromsome, a hyphenated range, or comma separated list.' optional='true'/>
```
Comment (bgruening) [inline tools/plink/plink.xml:499]: adding a validator here and exclude strange characters?

File: tools/plink/plink.xml [hunk 4]
```diff
@@ -0,0 +1,1241 @@
...
+                                <option value='variant'>Around/between specific variant(s)</option>
+                                <option value='window'>Around a specific variant</option>
+                                <option value='range'>Within a specific area (Must also specify a single chromosome in above input)</option>
+                            </param>
+                            <when value='variant'>
+                                <param name='from' type='text' label='From' help='Variant ID. Excludes all variants on different chromosomes than the named variant, as well as those with smaller base-pair position values. If they are used together but the --from variant is after the --to variant, they are automatically swapped.' optional='true'/>
+                                <param name='to' type='text' label='To' help='Variant ID. Excludes all variants on different chromosomes than the named variant, as well as those with larger base-pair position values. If they are used together but the --from variant is after the --to variant, they are automatically swapped.' optional='true'/>
```
Comment (bgruening) [inline tools/plink/plink.xml:528]: help text is the same? Not sure what these parameters are doing...

File: tools/plink/plink.xml [hunk 5]
```diff
@@ -0,0 +1,1241 @@
...
+                            </when>
+                            <when value='window'>
+                                <param name='snp' type='text' label='SNP variant id to include' help='Use this OR SNP variant to exclude' optional='true'/>
+                                <param name='exclude_snp' type='text' label='SNP variant id to exclude' help='Use this OR SNP variant to include' optional='true'/>
```
Comment (bgruening) [inline tools/plink/plink.xml:532]: validator?

File: tools/plink/plink.xml [hunk 6]
```diff
@@ -0,0 +1,1241 @@
...
+                                <param name='to_bp' type='integer' label='to-bp' help='These flags let you use physical positions to specify a variant range to load. Must also have specified a chromosome in above settings.' optional='true'/>
+                            </when>
+                        </conditional>
+                    </when>
+                    <when value='multi'>
+                        <param name='force_intersect' type='boolean' truevalue='--force-intersect' falsevalue='' label='Force intersect'  help='To reduce the potential for confusion, PLINK 2 normally errors out when multiple variant-inclusion filters 
+                        (--extract[-intersect], --extract-col-cond, --from/--to, --from-bp/--to-bp, --snp, --snps) are specified, since it may not be obvious whether the intersection or union will be taken. 
+                        --force-intersect allows the run to proceed; the set intersection will be taken.'/>
+                        <param name='snps' type='text' label='List of SNP variant ids to include' optional='true'/>
```
Comment (bgruening) [inline tools/plink/plink.xml:545]: validator ... maybe a SNP validator macro?

File: tools/plink/plink.xml (hunk 1 above)
Comment (astrovsky01) [reply tools/plink/plink.xml:1]: Yeah, thanks, missed that when changing it over

File: tools/plink/plink.xml (hunk 2 above)
Comment (astrovsky01) [reply tools/plink/plink.xml:102]: Nope -- specifically is --snps-only 'just-acgt'

It's functionally equivalent to an if with 2 elif, but shorter

**Comment by astrovsky01** [comment]
Found a problem with one of the filters in stats

File: tools/plink/plink.xml (hunk 4 above)
Comment (astrovsky01) [reply tools/plink/plink.xml:496]: Smaller vs larger

**Review by davebx — APPROVED** [review]

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @astrovsky01! This is awesome!

ping @blankenberg - you seemed to be interested in this tool a few years back

**Pull request merged**